public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/PartListSection.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { css } from '@emotion/css';
+import cx from 'classnames';
+import { SegmentAsync, Segment, SegmentInput, MenuItem, WithContextMenu, MenuGroup } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import { unwrap } from './unwrap';
+import { toSelectableValue } from './toSelectableValue';
+
+export type PartParams = Array<{
+  value: string;
+  options: (() => Promise<string[]>) | null;
+}>;
+
+type Props = {
+  parts: Array<{
+    name: string;
+    params: PartParams;
+  }>;
+  newPartOptions: SelectableValue[];
+  onChange: (partIndex: number, paramValues: string[]) => void;
+  onRemovePart: (index: number) => void;
+  onAddNewPart: (type: string) => void;
+};
+
+const renderRemovableNameMenuItems = (onClick: () => void) => {
+  return (
+    <MenuGroup label="" ariaLabel="">
+      <MenuItem label="remove" ariaLabel="remove" onClick={onClick} />
+    </MenuGroup>
+  );
+};
+
+const RemovableName = ({ name, onRemove }: { name: string; onRemove: () => void }) => {
+  return (
+    <WithContextMenu renderMenuItems={() => renderRemovableNameMenuItems(onRemove)}>
+      {({ openMenu }) => (
+        <button className="gf-form-label" onClick={openMenu}>
+          {name}
+        </button>
+      )}
+    </WithContextMenu>
+  );
+};
+
+type PartProps = {
+  name: string;
+  params: PartParams;
+  onRemove: () => void;
+  onChange: (paramValues: string[]) => void;
+};
+
+const noLeftPaddingClass = css({
+  paddingLeft: '0',
+});
+
+const Part = ({ name, params, onChange, onRemove }: PartProps): JSX.Element => {
+  const onParamChange = (par: string, i: number) => {
+    const newParams = params.map((p) => p.value);
+    newParams[i] = par;
+    onChange(newParams);
+  };
+  return (
+    <div className="gf-form-inline">
+      <div className={cx('gf-form-label', noLeftPaddingClass)}>
+        <RemovableName name={name} onRemove={onRemove} />(
+        {params.map((p, i) => {
+          const { value, options } = p;
+          return options != null ? (
```

Comment:
Is it `!=` on purpose instead of `!==` (e.g. to match `undefined` and `null`)?

Reply:
i changed it to `!==`... storytime: a long time ago i was using Facebook's Flow type-system, it's very similar to Typescript, and for certain reasons there, i got used to do the `!= null` to do, what you are also saying, to check for null-and-undef at the same time. but probably it's better to explictly check for null.